core/src/stage.cpp

```diff
@@ -212,6 +221,31 @@ void StagePrivate::newSolution(const SolutionBasePtr& solution) {
 		parent()->onNewSolution(*solution);
 }
 
+bool StagePrivate::addCost(SolutionBase& solution) {
+	double cost{ 0.0 };
+
+	auto* trajectory{ dynamic_cast<const SubTrajectory*>(&solution) };
+	if (trajectory && cost_term_) {
```

Comment:
Can we agree that each stage must have a `cost_term_`?
No need to check for its validity here then.

Reply:
Why do you say "Exactly" when you contradict my whole statement :question:

> because the solution has no costs associated initially

Of course they do! The user can already add a cost before calling `Stage::spawn` et al.
This is the only approach to set a cost prior to this work and I don't see a reason to remove that option.
Otherwise users would not only write custom Stages, but also custom CostTerms, that will only work with this one stage.
I only added explicit CostTerms for the standard stages to have them defined consistently.